src/components/PeopleTable.tsx

```diff
@@ -1,645 +1,292 @@
+type SortByWhat = 'sex' | 'born' | 'died' | 'name';
+
 export const PeopleTable = () => {
+  const [searchParams] = useSearchParams();
+  const [people, setPeople] = useState<People[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const { pathname } = useLocation();
+  const [filteredPeople, setFilteredPeople] = useState<People[]>([]);
+  const [currentIsReverse, setCurrentIsReverse] = useState<boolean>();
+  const [currentSortParam, setCurrentSortParam] = useState('');
+  const [errorText, setErrorTest] = useState('');
+  const getHref = (name: string) => {
+    return people.find(person => person.name === name)?.slug
+      ? people.find(person => person.name === name)?.slug
+      : '';
+  };
+
+  const getLink = (sortByWhat: SortByWhat) => {
+    const newSearchParams = new URLSearchParams(searchParams);
+
+    // Если уже сортируем по этому полю, переключаем порядок сортировки
+    if (currentSortParam === sortByWhat) {
+      const isCurrentlyDesc = newSearchParams.get('order') === 'desc';
+
+      // Переключаем порядок сортировки
+      if (isCurrentlyDesc) {
+        newSearchParams.delete('order'); // Если `desc`, удаляем для `asc`
+      } else {
+        newSearchParams.set('order', 'desc'); // Если `asc`, ставим `desc`
+      }
+    } else {
+      // Если сортируем по новому полю, задаем сортировку по возрастанию
+      newSearchParams.set('sort', sortByWhat);
+      newSearchParams.delete('order'); // Сбрасываем на `asc`
+    }
+
+    return `?${newSearchParams.toString()}`;
+  };
+
+  useEffect(() => {
+    getPeople()
+      .then(peopleArray => {
+        setPeople(peopleArray);
+        setFilteredPeople(peopleArray);
+        setIsLoading(false);
+      })
+      .catch(() => {
+        setErrorTest('Something went wrong');
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    const sexParam = searchParams.get('sex');
+    const sortParam = searchParams.get('sort');
+    const centuriesParams = searchParams.getAll('centuries').map(Number);
+    const queryParam = searchParams.get('query')?.toLowerCase() || '';
+    const isReverse = searchParams.get('order') === 'desc';
+
+    // Копируем исходный массив, чтобы начать с полного списка
+    const sortedPeople = [...people];
+
+    // Сортировка
+    if (sortParam) {
+      if (sortParam === 'sex') {
+        sortedPeople.sort((a, b) => a.sex.localeCompare(b.sex));
+      } else if (sortParam === 'name') {
+        sortedPeople.sort((a, b) => a.name.localeCompare(b.name));
+      } else if (sortParam === 'born') {
+        sortedPeople.sort((a, b) => a.born - b.born);
+      } else if (sortParam === 'died') {
+        sortedPeople.sort((a, b) => a.died - b.died);
+      }
+    }
+
+    // Если нужно отсортировать в обратном порядке
+    if (isReverse) {
+      sortedPeople.reverse();
+    }
+
+    // Фильтрация
+    const filtered = sortedPeople.filter(person => {
+      if (sexParam && person.sex !== (sexParam === 'male' ? 'm' : 'f')) {
```

Comment:
The condition for filtering by sex is incorrect. The sex values should be compared directly to 'male' or 'female', not 'm' or 'f'. Update the condition to `person.sex !== sexParam`.